Neos.ContentRepository.Core/Classes/Feature/Common/ConstraintChecks.php

```diff
@@ -139,6 +141,22 @@ protected function requireNodeTypeToNotBeOfTypeRoot(NodeType $nodeType): void
         }
     }
 
+    protected function requireRootNodeTypeToBeUnoccupied(
+        NodeTypeName $nodeTypeName,
+        ContentStreamId $contentStreamId,
+        ContentRepository $contentRepository
+    ): void {
+        try {
+            $rootNodeAggregate = $contentRepository->getContentGraph()->findRootNodeAggregateByType(
+                $contentStreamId,
+                $nodeTypeName
+            );
+            throw RootNodeAggregateTypeIsAlreadyOccupied::butWasExpectedNotTo($nodeTypeName);
+        } catch (RootNodeAggregateDoesNotExist $exception) {
+            // all is well
+        }
```

Comment:
empty `catch` blocks always make me suspicious (and in general: using exceptions for control flow).
With the change suggested above, this could be changed to:

```suggestion
        $rootNodeAggregate = $contentRepository->getContentGraph()->findRootNodeAggregateByType(
            $contentStreamId,
            $nodeTypeName
        );
        if ($rootNodeAggregate !== null) {
            throw RootNodeAggregateTypeIsAlreadyOccupied::butWasExpectedNotTo($nodeTypeName);
        }
```